walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -0,0 +1,68 @@
+package com.google.job.data;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.firestore.FireStoreUtils;
+
+import java.util.concurrent.ExecutionException;
+
+public final class JobsDatabase {
+    // TODO
+    private static final String COLLECTION = "Jobs";
+    private final Firestore firestore;
+
+    public JobsDatabase(Firestore firestore) {
+        this.firestore = firestore;
+    }
+
+    /**
+     * Adds a newly created job post.
+     *
+     * @param newJob Newly created job post.
+     * @return Id for the job in the database.
+     * @throws ExecutionException If error occurs when getting job id.
+     * @throws InterruptedException If error occurs when getting job id.
+     */
+    public String addJob(Job newJob) throws ExecutionException, InterruptedException {
+        ApiFuture<DocumentReference> addedDocRef = FireStoreUtils.store(COLLECTION, newJob);
```

Comment:
will we check the newJob params (that they are valid) before we send it to this function?

Reply:
I will add the comment to assume it is non-nullable; we need to check at the server level before sending it to this function